bika/lims/browser/sample/view.py

```diff
@@ -380,32 +388,37 @@ def folderitem(self, obj, item, index):
         obj = item['obj']
 
         item['replace']['getSampleID'] = "<a href='%s'>%s</a>" % \
-            (item['url'], obj.getSampleID())
+            (item['url'], obj.getSampleID)
+        analysis_requests_brains = arc(portal_type='AnalysisRequest', UID=obj.getAnalysisRequestsUID)
+        # From Professional Plone 4 development:
+        # The getURL() method returns, as with the absolute_url() method, the current referenced
+        # object URL. This may be different from the server URL at the time that the object was indexed
         item['replace']['Requests'] = ",".join(
-            ["<a href='%s'>%s</a>" % (o.absolute_url(), o.Title())
-             for o in obj.getAnalysisRequests()])
-        item['Client'] = obj.aq_parent.Title()
+            ["<a href='%s'>%s</a>" % (o.getURL(), o.Title)
+             for o in analysis_requests_brains])
+        item['Client'] = obj.getClientTitle
         if hideclientlink == False:
+            head, tail = os.path.split(obj.getURL())
             item['replace']['Client'] = "<a href='%s'>%s</a>" % \
-                (obj.aq_parent.absolute_url(), obj.aq_parent.Title())
-        item['Creator'] = self.user_fullname(obj.Creator())
+                (head, obj.getClientTitle)
+        item['Creator'] = self.user_fullname(obj.Creator)
 
-        item['DateReceived'] = self.ulocalized_time(obj.getDateReceived())
+        item['DateReceived'] = self.ulocalized_time(obj.getDateReceived)
 
-        deviation = obj.getSamplingDeviation()
-        item['SamplingDeviation'] = deviation and deviation.Title() or ''
+        item['SamplingDeviation'] = obj.getSamplingDeviationTitle
 
-        item['getStorageLocation'] = obj.getStorageLocation() and obj.getStorageLocation().Title() or ''
-        item['AdHoc'] = obj.getAdHoc() and True or ''
+        item['getStorageLocation'] = obj.getStorageLocationTitle
+        item['AdHoc'] = obj.getAdHoc
 
-        item['Created'] = self.ulocalized_time(obj.created())
+        item['Created'] = self.ulocalized_time(obj.created)
 
-        sd = obj.getSamplingDate()
+        sd = obj.getSamplingDate
         item['SamplingDate'] = \
             self.ulocalized_time(sd, long_format=1) if sd else ''
 
         after_icons = ''
-        if obj.getSampleType().getHazardous():
+        sample_type = bsc(portal_type='SampleType', UID=obj.getSampleTypeUID)[0]
```

Comment:
I would add a metadata in Sample's catalog for `hazardous`, so the additional query to retrieve the SampleType would not longer be required